From a93b56fc57b517f228910355d3a0fe8ff22ebc41 Mon Sep 17 00:00:00 2001
From: Srinivas Pandruvada <srinivas.pandruvada@linux.intel.com>
Date: Mon, 3 Feb 2014 10:26:13 -0800
Subject: [PATCH 3/3] Fix compenstion using "Processor" thermal cooling device

When any of the processor state is changed, it will impact all
processors. If thermal sysfs defines different cdev trip for
each processor, if we read and apply, the next processor will
already impacted by change by previous cdev.
---
 src/thd_cdev_therm_sys_fs.cpp | 7 +++++++
 1 file changed, 7 insertions(+)

diff --git a/src/thd_cdev_therm_sys_fs.cpp b/src/thd_cdev_therm_sys_fs.cpp
index b7d16f8..a01d608 100644
--- a/src/thd_cdev_therm_sys_fs.cpp
+++ b/src/thd_cdev_therm_sys_fs.cpp
@@ -54,6 +54,13 @@ int cthd_sysfs_cdev::update() {
 	tc_type_dev << "cooling_device" << index << "/type";
 	if (cdev_sysfs.exists(tc_type_dev.str())) {
 		cdev_sysfs.read(tc_type_dev.str(), type_str);
+		if (type_str.size()) {
+			// They essentially change same ACPI object, so reading their
+			// state from sysfs after a change to any processor will cause
+			// double compensation
+			if (type_str == "Processor")
+				read_back = false;
+		}
 	}
 
 	thd_log_debug("cooling dev %d:%d:%d:%s\n", index, curr_state, max_state,
-- 
1.9.rc1

